jax/experimental/checkify.py

```diff
@@ -212,26 +217,71 @@ def checkify_fun_to_jaxpr(f, error, in_avals):
 
 ## assert primitive
 
-def assert_(pred: Bool, msg: str) -> None:
+def assert_that(pred: Bool, msg: str) -> None:
+  """Assert a predicate, add an error with msg if predicate evaluates to False.
+
+  The predicate needs to be a jnp boolean.
```

Comment:
I think Python booleans work too! Maybe 'needs to be a boolean or rank-0 array with boolean dtype'? (Or am I missing something?)

Reply:
True! I wanted to make sure it mentions that you have to use `jnp` on traced values (to not get ConcretizationErrors) but that's not the same as requiring `jnp` bools.